src/set_theory/ordinal/basic.lean

```diff
@@ -494,23 +494,15 @@ def type (r : α → α → Prop) [wo : is_well_order α r] : ordinal :=
 def typein (r : α → α → Prop) [is_well_order α r] (a : α) : ordinal :=
 type (subrel r {b | r b a})
 
-theorem type_def (r : α → α → Prop) [wo : is_well_order α r] :
-  @eq ordinal ⟦⟨α, r, wo⟩⟧ (type r) := rfl
-
-@[simp] theorem type_def' (r : α → α → Prop) [is_well_order α r] {wo} :
-  @eq ordinal ⟦⟨α, r, wo⟩⟧ (type r) := rfl
+@[simp] theorem type_def (w : Well_order) : ⟦w⟧ = type w.r :=
+by { cases w, refl }
```

Comment:
I think it's worth having both lemmas, since the new lemmas can't be used by `dsimp`, but the old one can.

Reply:
They're certainly defeq since they're both proved by `rfl`. I can only speculate that there's some weird syntactic matching going on that means both are needed, but I don't really know.

Reply:
Well, neither of them was even used before this PR, so I feel confident in just keeping one of the originals and my new lemma.